src/core/interfaces.ts

```diff
@@ -476,6 +476,29 @@ export namespace CoreChartProps {
   export type I18nStrings = CartesianI18nStrings & PieI18nStrings;
 }
 
+export interface CoreLegendProps {
+  series: Highcharts.SeriesOptionsType[];
+  title?: string;
+  actions?: React.ReactNode;
+  position?: "bottom" | "side";
+  i18nStrings?: BaseI18nStrings;
+  onClearHighlight?: NonCancelableEventHandler<CoreChartProps.HighlightClearDetail>;
+  onItemHighlight?: NonCancelableEventHandler<CoreChartProps.LegendItemHighlightDetail>;
+  onVisibleItemsChange?: NonCancelableEventHandler<CoreChartProps.VisibleItemsChangeDetail>;
+  getLegendTooltipContent?: CoreChartProps.GetLegendTooltipContent;
+  callback?: (api: CoreLegendProps.LegendAPI) => void;
+}
+
+export namespace CoreLegendProps {
```

Comment:
The namespace for the legend should include all relevant props. So e.g. instead of reusing the LegendItem interface from the CoreChart namespace, we should proxy it here, e.g.:

```
export interface CoreLegendProps {
  items: readonly CoreLegendProps.LegendItem[];
  // ...
}

export namespace CoreLegendProps {
   export type LegendItem = CoreChartProps.LegendItem[]; // better do vice-versa, so that the chart imports from the legend
   // ...
}
```